### PR TITLE
this was something quick

### PR DESCRIPTION
**Pull request by Buttonz52**
Alright this needs MAJOR testing with memSlice (mikes code).

### CODE REVIEW

File: RayTrace.cpp [hunk 1]
```diff
@@ -0,0 +1,252 @@
+#include <glm/glm.hpp>
+#include <vector>
+#include <iostream>
+#include <stdio.h>
+
+using namespace glm;
+using namespace std;
+
+struct Triangle
+{
```
Comment (cwoz117) [inline RayTrace.cpp]: WTF

File: RayTrace.cpp [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+
+	getInfo(sharedBuffer); // assumes [header,scene,pixel_pos]
+
+	for (int i = 0; i < pixel_pos.size(); i++)
+	{
+		vec3 c = ray_trace(pixel_pos[i].x, pixel_pos[i].y);
+		colors.push_back(c);
+	}
+
+	cout << &colors[0];
```
Comment (cwoz117) [inline RayTrace.cpp]: push the buffer to cout to avoid shared memory (?)

File: RayTrace.cpp [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+	{
+		vec3 ppco = p.position - r.origin; //planePosition -cameraOrigin
+		float t = dot(ppco, normal) / para;
+
+		if (t < 0)
+			return vec4(NULL, NULL, NULL, NULL);
+
+		vec3 x = r.origin + (t*r.direction); //plane.w is t
+
+											 //then calc if point is on triangle
```
Comment (cwoz117) [inline RayTrace.cpp]: clean whitespace

File: RayTrace.cpp (hunk 2 above)
Comment (Buttonz52) [inline RayTrace.cpp]: Not sure what that looks like, as colors is a vector<vec3> not a buffer

**Pull request merged**